src/views/ManageList.jsx

```diff
@@ -77,3 +79,69 @@ export function ManageList({ listPath }) {
 		</form>
 	);
 }
+
+function ShareList({ listPath }) {
+	const [recipientEmail, setRecipientEmail] = useState('');
+	const [emailValidator, setEmailValidator] = useState('');
+	const [isValid, setIsValid] = useState(false);
+	const { user } = useAuth();
+	const currentUserId = user?.uid;
+	console.log(currentUserId);
+
+	const handleChange = (e) => {
+		setRecipientEmail(e.target.value);
+
+		if (validateEmail(e.target.value)) {
+			setEmailValidator('');
+			setIsValid(true);
+		} else {
+			setEmailValidator('Email is not valid');
+		}
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		if (recipientEmail === '') {
+			return alert('Cannot submit without an email.');
+		}
+
+		if (!isValid) {
+			return alert('Cannot submit with an invalid email');
+		}
+
+		try {
+			await shareList(listPath, currentUserId, recipientEmail);
+			alert('List was successfully shared!');
+			setRecipientEmail('');
+		} catch (error) {
+			alert(error);
+		}
+	};
+
+	return (
+		<form id="shareListForm" onSubmit={handleSubmit}>
```

Comment:
Do we need a form id here? Looks like we don't use it.

Reply:
Agreed, we have removed the id.